### ISSUE TITLE
`require-tagless-components` rule fails for `Service.extend`

### ISSUE THREAD

**Issue by mehulkar** [ember-cli/eslint-plugin-ember#601]
In `tests/integration/components/foo-test.js`, if I have:

```
import Service from '@ember/service';
Service.extend({});
```

the lint rule fails

**Comment by bmish** [comment]
Can you include the error it fails with?

CC: @alexlafroscia

**Comment by mehulkar** [comment]
Seems similar to #235, and based on an initial look, looks like the problem is here: https://github.com/ember-cli/eslint-plugin-ember/blob/master/lib/utils/ember.js#L171-L192

**Comment by mehulkar** [comment]
@bmish do you mean the lint failure error message?  (I'm not getting an exception thrown, just a false positive on the lint rule)

```
113:21  error  Please switch to a tagless component by setting `tagName: ''` or converting to a Glimmer component  ember/require-tagless-components
```

**Comment by alexlafroscia** [comment]
Ah, I suppose we missed test cases for Ember objects that are not components! I can take a look at fixing this tonight.

**Comment by alexlafroscia** [comment]
I added a couple of test cases that look like your example @mehulkar -- however, the tests don't actually report the false positive that you're seeing.

Are you running the latest version of `eslint-plugin-ember`?

**Comment by bmish** [comment]
Try specifying a component filename as mentioned in the ticket.

**Comment by alexlafroscia** [comment]
Oooh yup, that's what it was!

It _really_ feels like getting away from file names as part of the lint rules would be good, and instead just looking at the actual source of the import 😅

**Comment by bmish** [comment]
Yeah, definitely agree that we should be checking imports instead of file paths, tracking that improvement with #590.

**Comment by alexlafroscia** [comment]
Would you be OK with a PR that fixed that? It would be pretty extensive, but I think that's really the _right_ fix in this case.

**Comment by bmish** [comment]
Yeah, since this lint rule just relies on `emberUtils.isEmberComponent()`, I agree that the right fix would be to check imports instead of filenames inside that util function. That would affect a lot of rules, but I would support pursuing that fix.

**Comment by alexlafroscia** [comment]
Sweet! I'm going to take a stab at this, I have some time I can invest in it over the next few days

**Comment by Turbo87** [comment]
@mehulkar @alexlafroscia is this still an issue or has it been fixed by #603?

**Comment by alexlafroscia** [comment]
That PR was never merged, and at this point I can't see myself coming back to it unfortunately. I just don't have much OSS time these days.

**Comment by Turbo87** [comment]
oh, lol, I totally missed that the PR was still open 😅

**Comment by ballPointPenguin** [comment]
I see false positives as well when lodash `extend` is in use within a component, eg

```
_.extend(notAComponentOrClass(), { foo: 'bar' })
```
